actions/class.Runner.php

```diff
@@ -1065,4 +1040,80 @@ protected function getRawRequestParameter($name)
         }
         return $parameters[$name];
     }
+
+    /**
+     * @param QtiRunnerServiceContext $serviceContext
+     * @return array
+     * @throws QtiRunnerClosedException
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     * @throws \qtism\runtime\storage\common\StorageException
+     * @throws common_Exception
+     * @throws common_exception_Error
+     * @throws common_exception_InvalidArgumentType
+     * @throws common_ext_ExtensionException
+     */
+    protected function getInitResponse($serviceContext)
+    {
+        if (
+            $this->hasRequestParameter('clientState')
+            && $this->getRequestParameter('clientState') === 'paused'
+        ) {
+            $this->getRunnerService()->pause($serviceContext);
+            $this->getRunnerService()->check($serviceContext);
+        }
+
+        $result = $this->getRunnerService()->init($serviceContext);
+        $this->getRunnerService()->persist($serviceContext);
+
+        return $this->getInitSerializedResponse($result, $serviceContext);
+    }
+
+    /**
+     * Checks the storeId request parameter and returns the last store id if set, false otherwise
+     *
+     * @param QtiRunnerServiceContext $serviceContext
+     * @return string|boolean
+     * @throws common_exception_InvalidArgumentType
+     */
+    private function getClientStoreId($serviceContext)
+    {
+        if (
+            $this->hasRequestParameter('storeId')
+            && preg_match('/^[a-z0-9\-]+$/i', $this->getRequestParameter('storeId'))
+        ) {
+            return $this->getRunnerService()->switchClientStoreId(
+                $serviceContext,
+                $this->getRequestParameter('storeId')
+            );
+        }
+
+        return false;
+    }
+
+    /**
+     * @param bool $isInitialized
+     * @param QtiRunnerServiceContext $serviceContext
+     * @return array
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     * @throws common_Exception
+     * @throws common_exception_InvalidArgumentType
+     * @throws common_ext_ExtensionException
+     */
+    private function getInitSerializedResponse($isInitialized, $serviceContext)
```

Comment:
I would remove the flag parameter and generate only the successful response here. Opinion?

Reply:
Fixed.